pulp_container/app/registry_api.py

```diff
@@ -1036,10 +1039,11 @@ def handle_safe_method(self, request, path, pk):
                 tag = models.Tag.objects.get(name=pk, pk__in=repository_version.content)
             except models.Tag.DoesNotExist:
                 distribution = distribution.cast()
+                permission_checker = PermissionChecker(request.user)
                 if (
                     distribution.remote
-                    and distribution.pull_through_distribution_id
-                    and request.user.is_authenticated
+                    and distribution.pull_through_distribution
+                    and permission_checker.has_pull_through_permissions(distribution)
```

Comment:
This (and a similar one below) is the core of the change, right?

Reply:
Yes, once we pass through the registry token steps and a user can access the particular endpoint, we should finally ensure that users without valid permissions should not pull new content into a repository. But, they can still pull existing content.

Reply:
This granularity cannot be achieved purely with token authentication. That is why I had to introduce this check.